test/index.js

```diff
@@ -765,12 +765,48 @@ function suite(moduleName) {
             dir +
             '/fixtures/symlinks/symlink-dest/hey/isaidhey/whatsgoingon/test.txt',
         },
+
+        {
+          cwd: dir,
+          base: dir + '/fixtures/symlinks',
+          path: dir + '/fixtures/symlinks/folder-a/folder-a-file.txt',
+        },
+        {
+          cwd: dir,
+          base: dir + '/fixtures/symlinks',
+          path: dir + '/fixtures/symlinks/folder-b/folder-b-file.txt',
+        },
+
+        // It should follow these circular symlinks, but not infinitely
+        {
+          cwd: dir,
+          base: dir + '/fixtures/symlinks',
+          path: dir + '/fixtures/symlinks/folder-a/link-to-b/folder-b-file.txt',
+        },
+        {
+          cwd: dir,
+          base: dir + '/fixtures/symlinks',
+          path: dir + '/fixtures/symlinks/folder-b/link-to-a/folder-a-file.txt',
+        },
+
+        // And it should follow a symlink to a parent directory (circular symlink) without blowing up
+        {
+          cwd: dir,
+          base: dir + '/fixtures/symlinks',
+          path:
+            dir +
+            '/fixtures/symlinks/symlink-dest/hey/isaidhey/whatsgoingon/test.txt',
+        },
```

Comment:
I need to check if this is a duplicate glob getting through because we wouldn't want to emit it twice.